editor/editor_inspector.cpp

```diff
@@ -681,9 +681,45 @@ bool EditorProperty::is_selected() const {
 	return selected;
 }
 
+void EditorProperty::set_disabled(bool p_disabled) {
```

Comment:
I don't think this is that right way to do it. It's the simplest way I suppose, but it's still quite hacky.
I believe think should be made a virtual function, implemented differently for each class extending EditorProperty.

In most cases, this is going to be about simply disabling the input field, which should not be that hard to do.

Reply:
The implementation should now be more in line with what you suggested.